Assets/Tests/InputSystem/CoreTests_GlobalActions.cs

```diff
@@ -0,0 +1,500 @@
+        action.AddBinding("<keyboard>/b", interactions: "tap");
+        var input = new Input<float>(action);
+
+        Assert.That(input.GetInteraction<HoldInteraction>(input.bindings[0]).isValid, Is.True);
+        Assert.That(input.GetInteraction<TapInteraction>(input.bindings[0]).isValid, Is.False);
+
+        Assert.That(input.GetInteraction<HoldInteraction>(input.bindings[1]).isValid, Is.False);
+        Assert.That(input.GetInteraction<TapInteraction>(input.bindings[1]).isValid, Is.True);
+    }
+
+    [Test]
+    [Category(TestCategory)]
+    public void GlobalActions_InteractionIsValid_WhenNoBindingIndexIsSpecified()
+    {
+        var asset = ScriptableObject.CreateInstance<InputActionAsset>();
+        asset.AddActionMap("ActionMapOne").AddBinding("<keyboard>/x");
+        var map = asset.AddActionMap("ActionMapTwo");
+        var action = map.AddAction("ActionOne", binding: "<keyboard>/a", interactions: "hold");
+        action.AddBinding("<keyboard>/b", interactions: "tap");
+        var input = new Input<float>(action);
+
+        Assert.That(input.GetInteraction<HoldInteraction>(BindingIndex.None).isValid, Is.True);
+        Assert.That(input.GetInteraction<TapInteraction>(BindingIndex.None).isValid, Is.True);
+    }
+
+    [Test]
+    [Category(TestCategory)]
+    public void GlobalActions_InteractionModificationsSurviveFullBindingReResolution()
+    {
+        var map = new InputActionMap();
+        var action = map.AddAction("Action", binding: "<keyboard>/a", interactions: "hold(duration=0.5)");
+
+        var input = new Input<float>(action);
+        input.SetInteractionParameter<HoldInteraction, float>(input.bindings[0], x => x.duration, 1.5f);
+
+        // adding a binding causes a full binding resolution
+        action.AddBinding("<Gamepad>/rightTrigger");
+
+        Assert.That(input.GetInteractionParameter<HoldInteraction, float>(input.bindings[0], x => x.duration), Is.EqualTo(1.5f));
+    }
+
+    [Test]
+    [Category(TestCategory)]
+    [TestCase(InputBindingsCollection.EnumerationBehaviour.SkipCompositeParts)]
+    [TestCase(InputBindingsCollection.EnumerationBehaviour.IncludeCompositeParts)]
+    public void GlobalActions_InputBindingsCollection_CanIterateDependingOnMode(InputBindingsCollection.EnumerationBehaviour enumerationBehaviour)
+    {
+        var action = new InputAction(binding: "<keyboard>/w");
+        action.AddBinding("<keyboard>/a");
+        action.AddCompositeBinding("OneModifier")
+            .With("modifier", "<keyboard>/leftShift")
+            .With("binding", "<keyboard>/a");
+        action.AddBinding("<gamepad>/leftStick");
+        action.AddCompositeBinding("2DVector")
+            .With("up", "<keyboard>/w")
+            .With("down", "<keyboard>/s")
+            .With("left", "<keyboard>/a")
+            .With("right", "<keyboard>/d");
+
+        var collection = new InputBindingsCollection(action, enumerationBehaviour);
```

Comment:
Seems to be testing newly added concept which is good to see.

The indexing below 0,1,2,5,6 isn't obvious to me though.

0 seems to be w
1 seems to be a
2,3,4 are composites (OneModifier, leftstick, a)
5 seems to be gamepad left stick
I would expect 6,7,8,9,10 to be composites  (2dvector, w,s,a,d)

so I am puzzled with the index 6 in the 'skipcompositeparts' list